### PR TITLE
Latexlab/uf review 03

### PR DESCRIPTION
**Pull request by josephwright**
**READ ME FIRST**: Please understand that in most cases we will not be able to merge a pull request because there are a lot of internal activities needed when updating the LaTeX2e sources. If you have a code suggestion please discuss it with the team first.

*Pull requests in this repository are intended for LaTeX Team members only.*

# Internal housekeeping

## Status of pull request

- Feedback wanted 
- Under development
-[x] Ready to merge

### CODE REVIEW

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (josephwright) [inline required/latex-lab/latex-lab-footnotes.dtx:1730]: `:ee` in the next diff ;)

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-footnotes.dtx:1856]: :ee doesn't seem to make sense here and only costs processing cycles, should be :en I guess

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-footnotes.dtx:1862]: :Nx -> :Ne   and :ee -> :en ?

File: required/latex-lab/latex-lab-graphic.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-graphic.dtx:190]: maybe actually explain the reason for this rather than referring to zref-savepos

File: required/latex-lab/latex-lab-toc-kernel-changes.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-toc-kernel-changes.dtx:157]: I really prefer a few  `%` lines (all over the place not just here) to make it easier to see where sections start in the source.

File: required/latex-lab/latex-lab-toc-kernel-changes.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-toc-kernel-changes.dtx:160]: would work

File: required/latex-lab/latex-lab-toc.dtx
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-toc.dtx:118]: set up

**Review by FrankMittelbach — CHANGES_REQUESTED** [review]
a few comments and suggestions for change, but all minor I guess.

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (josephwright) [reply required/latex-lab/latex-lab-footnotes.dtx:1862]: I think we are holding off on `\tl_set:Nx`, etc., for a bit as we want to avoid `expl3` issues as far as possible for users

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (u-fischer) [reply required/latex-lab/latex-lab-footnotes.dtx:1856]: there is no `\property_ref:en`, so the question is if one should generate a variant for this?

Comment (FrankMittelbach) [reply required/latex-lab/latex-lab-footnotes.dtx:1856]: I think yes. After all executing "e" on somehting that is already a fixed string is just costing execution time.

File: required/latex-lab/latex-lab-footnotes.dtx
Comment (FrankMittelbach) [reply required/latex-lab/latex-lab-footnotes.dtx]: @josephwright while that might be an issue with normal packages, I don't see the need for latex-lab code. That usually requires a current L3 prog layer anyway, I would have thought.

**Review by FrankMittelbach — APPROVED** [review]

**Comment by u-fischer** [comment]
well switching to e-type is hard. The branch here fails because there was still an x-type in ltproperties. I have changed that now in the ltproperties-activechars branch and once that pass will merge that into develop and then repair the tests here.

**Pull request merged**
